cas-server-support-x509/src/main/java/org/jasig/cas/adaptors/x509/authentication/handler/support/CRLDistributionPointRevocationChecker.java

```diff
@@ -199,16 +199,22 @@ protected boolean addCRL(final Object id, final X509CRL crl) {
         }
 
         final List<URI> urls = new ArrayList<>();
-        for (final DistributionPoint point : points.getItems()) {
-            final Object location = point.getDistributionPoint();
-            if (location instanceof String) {
-                addURL(urls, (String) location);
-            } else if (location instanceof GeneralNameList) {
-                for (final GeneralName gn : ((GeneralNameList) location).getItems()) {
-                    addURL(urls, gn.getName());
+        
+        if(points != null){
+            for (final DistributionPoint point : points) {
+                final DistributionPointName pointName = point.getDistributionPoint();
+                if(pointName != null){
+                    final ASN1Sequence nameSequence = ASN1Sequence.getInstance(pointName.getName());
+                    for (int i = 0; i < nameSequence.size(); i++) {
+                        final GeneralName name = GeneralName.getInstance(nameSequence.getObjectAt(i));
+                        logger.debug("Found CRL distribution point {}.", name);
+                        try {
+                            addURL(urls, DERIA5String.getInstance(name.getName()).getString());
+                        } catch (final Exception e) {
```

Comment:
In the first try catch block can be thrown IllegalArgumentException, for example, due to invalid distribution point, or when Cryptacular tries to read the data it can throw IllegalArgumentException due to an IOException when exception parsing ASN.1 data.